jobsdb/unionQuery.go

```diff
@@ -0,0 +1,284 @@
+package jobsdb
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+const (
+	selectQuery = `SELECT jobs.job_id, jobs.uuid, jobs.user_id, jobs.parameters, jobs.custom_val, jobs.event_payload, jobs.event_count, jobs.created_at, jobs.expire_at, jobs.customer, sum(jobs.event_count) over (order by jobs.job_id asc) as running_event_counts `
+)
+
+//used to get pickup-counts during server start-up
+func (jd *HandleT) GetCustomerCounts(defaultBatchSize int) map[string]int {
+	unionMap := make(map[string]int)
+	//just using first DS here
+	//get all jobs all over DSs and then calculate
+	//use DSListLock also
+
+	// jd.dsMigrationLock.RLock()
+	jd.dsListLock.RLock()
+	// defer jd.dsMigrationLock.RUnlock()
+	defer jd.dsListLock.RUnlock()
+	rows, err := jd.dbHandle.Query(fmt.Sprintf(`select customer, count(job_id) from %s group by customer;`, jd.getDSList(false)[0].JobTable))
+	jd.assertError(err)
+
+	for rows.Next() {
+		var customer string
+		var count int
+		err := rows.Scan(&customer, &count)
+		jd.assertError(err)
+		unionMap[customer] = count
+	}
+
```

Comment:
https://golang.org/doc/database/querying#multiple_rows


They are two reasons  `.Next()` will return false and stop iteration:
1. All rows have been consumed 
2. An error took place while iterating

https://pkg.go.dev/database/sql#Rows.Next

For the second reason, once the iteration is over we need to check `.Err`

```suggestion
	}
    if err = rows.Err(); err != nil {
        jd.assertError(err)
    }
```

Maybe we need to introduce this check if it is missing.